app/main.py

```diff
@@ -1,3 +1,122 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(
+        self, other: Union["Distance", int, float]
+    ) -> Union["Distance", TypeError]:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            other = Distance(other)
+            return Distance(self.km + other.km)
+        raise TypeError(
+            f"unsupported type for __add__ method. 'Distance', "
+            f"<class 'float'> or <class 'int'> were expected. "
+            f"But received {type(other)}"
+        )
```

Comment:
if you had used type annotation from future you wouldn't have had to describe the type error in all this cases, as it would be pretty much the same by default.